ac_new.rb

```diff
@@ -1,388 +1,367 @@
-	
-    basedir=File.dirname(__FILE__) 
+    ffmpeg_bin = ENV['FFMPEG_BIN'] || 'ffmpeg'
+
+    ffprobe_bin = ENV['FFPROBE_BIN'] || 'ffprobe'
+
+    resolution = '1280x720'
+    frame_rate = '30'
+    audio_file = out_dir + meeting_id + '.mp3'
+    recording_file = out_dir + meeting_id + '.mkv'
+    full_recording_file = out_dir + meeting_id + '.full.mkv'
+
+    basedir = File.dirname(__FILE__)
     @logger.info(basedir + '/get_ac_audio.sh ' + meeting_id)
-    system basedir + '/get_ac_audio.sh ' + meeting_id 
-    if $?.exitstatus != 0 or ! File.exist?(audio_file)
-	    @logger.error("Failed to obtain audio file :(")
-	    return false
+    system basedir + '/get_ac_audio.sh ' + meeting_id
+    if ($CHILD_STATUS.exitstatus != 0) || !File.exist?(audio_file)
+      @logger.error('Failed to obtain audio file :(')
+      return false
     end
 
-    # get duration from the MP3 file, we'll use that to determine how long ffmpeg should be recording for 
-    dur_sec, stdeerr, status = Open3.capture3(ffprobe_bin + " -v error -show_entries format=duration -of default=noprint_wrappers=1:nokey=1 " + audio_file.shellescape)
-    dur_sec=dur_sec.delete!("\n")
-    if ! status.success?
-      log.error('Failed to get audio track duration. Exited with ' + $?.exitstatus.to_s + ':(')
+    # get duration from the MP3 file, we'll use that to determine how long ffmpeg should be recording for
+    dur_sec, stdeerr, status = Open3.capture3(ffprobe_bin + ' -v error -show_entries format=duration -of default=noprint_wrappers=1:nokey=1 ' + audio_file.shellescape)
+    dur_sec = dur_sec.delete!("\n")
+    unless status.success?
+      log.error('Failed to get audio track duration. Exited with ' + $CHILD_STATUS.exitstatus.to_s + ':(')
       return false
     end
     # since AC takes forever to load the recording, add 2 minutes to the actual recording's duration, we'll cut the extra off later
-    extra_duration=dur_sec.to_f + 120
+    extra_duration = dur_sec.to_f + 120
 
-    if ENV['AC_LOGIN_REQUIRED'] === "true"
-      @driver.get(@base_url + "/system/login?logintype=oldstyle&next=/admin")
-      @driver.find_element(:id, "name").send_keys ac_username
-      @driver.find_element(:id, "pwd").send_keys ac_passwd
-      @driver.find_element(:id, "login-button").click
+    if ENV['AC_LOGIN_REQUIRED'] === 'true'
+      @driver.get(@base_url + '/system/login?logintype=oldstyle&next=/admin')
+      @driver.find_element(:id, 'name').send_keys ac_username
+      @driver.find_element(:id, 'pwd').send_keys ac_passwd
+      @driver.find_element(:id, 'login-button').click
     end
-    @driver.get(@base_url + "/" + meeting_id +"?launcher=false&fcsContent=true&pbMode=normal")
+    @driver.get(@base_url + '/' + meeting_id + '?launcher=false&fcsContent=true&pbMode=normal')
 
     # FFmpeg magic
-    # record X11's display 
-    if ! ffmpeg_x11_grab(ffmpeg_bin,resolution, frame_rate, x_display, extra_duration.to_s, recording_file)
```

Comment:
RE all `if !` vs. `unless`: again, it's a coding style thing. While I sometimes use `unless` and other such syntactic sugar and one of my favourite interpreter langs is Perl which, one may argue is the queen of syntactic sugar, in this case, I prefer my original lines.

Reply:
No problem, ruby isn't my first language, I tend to be more JavaScript C# person :)